Submitted By: Mark Vels <mark.vels@fundem.net>
Date: 2007-07-20
Initial Package Version: 2.6
Origin: original patch glibc-2.6-mawk_fix-1.patch by Jim Gifford <jim@linuxfromscratch.org>
Upstream Status: Not submitted
Description: Fixes build if host distro is using mawk.
	     Distro's known to have this issue are Debian 
	     and Ubuntu

	     Changes wrt version 1: added two more cases.

diff -Naur glibc-2.6.orig/scripts/gen-sorted.awk glibc-2.6/scripts/gen-sorted.awk
--- glibc-2.6.orig/scripts/gen-sorted.awk	2006-02-28 07:05:57.000000000 +0000
+++ glibc-2.6/scripts/gen-sorted.awk	2007-07-20 21:03:50.000000000 +0000
@@ -16,7 +16,7 @@
 {
   subdir = type = FILENAME;
   sub(/^.*\//, "", type);
-  sub(/\/[^/]+$/, "", subdir);
+  sub(/\/[^\/]+$/, "", subdir);
   sub(/^.*\//, "", subdir);
   thisdir = "";
 }
@@ -56,13 +56,13 @@
     # The Subdirs file comes from an add-on that should have the subdirectory.
     dir = FILENAME;
     do
-      sub(/\/[^/]+$/, "", dir);
+      sub(/\/[^\/]+$/, "", dir);
     while (dir !~ /\/sysdeps$/);
     sub(/\/sysdeps$/, "", dir);
     if (system("test -d " dir "/" thisdir) == 0)
       dir = dir "/" thisdir;
     else {
-      sub(/\/[^/]+$/, "", dir);
+      sub(/\/[^\/]+$/, "", dir);
       if (system("test -d " dir "/" thisdir) == 0)
         dir = dir "/" thisdir;
       else {
